### PR TITLE
Chrome: Adding the Revisions Panel

### LINKED ISSUE #853: Sidebar: Link to latest revision
Design:

screen shot 2017-05-22 at 11 07 01

Clicking this link should simply take you to the latest revision of the post (example /wp-admin/revision.php?revision=23595). That page has buttons to navigate to previous revisions also. 

Context:

post settings

### PR DESCRIPTION
**Pull request by youknowriad**
closes #853 

 - The revisions are fetched in the component itself.
 - We trigger a new fetch each time the post is saved.
 - I extracted some URL utils to the `url` utils file.

### CODE REVIEW

File: editor/actions.js [hunk 1]
```diff
@@ -96,8 +95,11 @@ export function savePost( dispatch, postId, edits ) {
 
 export function trashPost( dispatch, postId, postType ) {
 	new wp.api.models.Post( { id: postId } ).destroy().done( () => {
-		window.location.href = 'edit.php?post_type=' + postType
-			+ '&trashed=1&ids=' + postId;
+		window.location.href = getWPAdminURL( 'edit.php', {
```
Comment (aduth) [inline editor/actions.js]: Could this just reuse `addQueryArgs`, e.g. `addQueryArgs( 'edit.php', { post_type: postType } );` ?

Comment (youknowriad) [inline editor/actions.js]: Yes probably, but I thought it should be a separate function to be able to change its implementation later (#867)

**Comment by jasmussen** [comment]
Designwise this looks great! And great to see all the progress on the sidebar as a whole!

I'm getting more and more JS errors with 405 though. This one using PUT:

screen shot 2017-05-30 at 06 42 13

@nylen any tips?

**Comment by youknowriad** [comment]
I'm merging soon if no objections?

File: editor/effects.js [hunk 2]
```diff
@@ -2,12 +2,12 @@
  * External dependencies
  */
 import { get } from 'lodash';
-import { parse, stringify } from 'querystring';
 
 /**
- * WordPress dependencies
+ * Internal dependencies
  */
 import { getBlockSettings, switchToBlockType } from 'blocks';
```
Comment (aduth) [inline editor/effects.js]: There's both WordPress and Internal dependencies here.

Aside: We could do for a better indicator.

File: editor/effects.js [hunk 3]
```diff
@@ -40,15 +41,11 @@ export default {
...
-		const qs = parse( query || '' );
-		if ( qs.post_id === post.id ) {
+		const { post, isNew } = action;
+		if ( ! isNew ) {
 			return;
 		}
-
-		const newUrl = baseUrl + '?' + stringify( {
-			...qs,
+		const newUrl = getGutenbergURL( {
```
Comment (aduth) [inline editor/effects.js]: Inconsistency between `Url` and `URL`. I think we'll want to settle on and use consistently a recommendation for acronyms. The core guidelines do not include one. Calypso recommends strict camel-casing, but I've personally grown fonder of upper-case. I can create an issue.

File: editor/utils/url.js [hunk 4]
```diff
@@ -18,3 +19,32 @@ export function addQueryArgs( url, args ) {
...
+/**
+ * Returns the Gutenberg page URL with extra query strings
+ *
+ * @param  {Object} query  Query Args
+ *
+ * @return {String}        URL
+ */
+export function getGutenbergURL( query = {} ) {
+	const [ baseUrl, currentQuery ] = window.location.href.split( '?' );
+	const qs = parseQueryString( currentQuery || '' );
```
Comment (aduth) [inline editor/utils/url.js]: Minor: The default can be assigned in the array destructure for clarity/consistency:

```js
const [ baseUrl, currentQuery = '' ] = window.location.href.split( '?' );
```

**Pull request merged**